### ISSUE TITLE
Pixie's OTEL exports tries to send non UTF-8 data as string protos

### ISSUE THREAD

**Issue by vihangm** [pixie-io/pixie#1594]
**Describe the bug**
The protobuf spec clearly states that `string` types must contain UTF-8 strings.
Pixie uses `bytes` for `StringColumn` since traced data might not always be valid UTF-8.
The OTel [spec](https://opentelemetry.io/docs/specs/otel/common/attribute-type-mapping/#string-values) also suggests using `string` or `bytes` depending on whether the data is valid UTF-8.

However the `Pixie` OTEL exporter uses `string` type when exporting columns that are `types::STRING`
See [here](https://github.com/pixie-io/pixie/blob/c092058a5172396ea9bead1e209c2ed4e2336943/src/carnot/exec/otel_export_sink_node.cc#L106-L109).

This causes errors like the following:
```
[libprotobuf ERROR external/com_google_protobuf/src/google/protobuf/wire_format_lite.cc:618] String field 'opentelemetry.proto.common.v1.AnyValue.string_value' contains invalid UTF-8 data when serializing a protocol buffer. Use the 'bytes' type if you intend to send raw bytes. 
```
This causes Pixie to drop such data during the export.

**Expected behavior**
Pixie should export data to OTel using the bytes type when there are non UTF-8 strings involved.

**Comment by rosenhouse** [comment]
Is this issue fixed by #1617?

**Comment by aimichelle** [comment]
Hey @rosenhouse !  Yup, it should be fixed by #1617 . 
Although there is one thing we were thinking about. Currently we'll send the data in the bytes field if it contains non-UTF-8 characters. However, it does seem annoying from the consumer-side to have to switch between bytes/string depending on the contents.
Would it actually be better for you all if we just send all data in bytes, assuming any could contain non-UTF-8.

**Comment by rosenhouse** [comment]
Skimming the OTLP specs, it sounds like you're doing the right thing now: https://opentelemetry.io/docs/specs/otel/common/attribute-type-mapping/#string-values

I guess that depends on how its treated by common OTLP servers.  We're using [opentelemetry-collector-contrib](https://github.com/open-telemetry/) and I know others who are using Telegraf.

We should probably add some tests to the later stages of our processing to cover this non-UTF-8 case.
